packages/web-react/src/components/FileUploader/useFileUploaderInput.ts

```diff
@@ -0,0 +1,131 @@
+import { ChangeEvent, DragEvent, useState } from 'react';
+import { useFileUploaderContext } from './FileUploaderContext';
+import {
+  DEFAULT_ERROR_MESSAGE_MAX_FILE_SIZE,
+  DEFAULT_ERROR_MESSAGE_QUEUE_DUPLICITY,
+  DEFAULT_ERROR_MESSAGE_QUEUE_LIMIT,
+} from './constants';
+
+export interface UseFileUploaderInputProps {
+  accept?: string;
+  fileLimitMaxSize: number;
+  fileQueueLimit: number;
+  isMultiple?: boolean;
+  onError?: (error: string) => void;
+}
+
+export interface UseFileUploaderInputReturn {
+  isDragging: boolean;
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  onDragEnter: () => void;
+  onDragLeave: () => void;
+  onDragOver: (event: DragEvent<HTMLDivElement>) => void;
+  onDrop: (event: DragEvent<HTMLDivElement>) => void;
+}
+
+export const useFileUploaderInput = (props: UseFileUploaderInputProps): UseFileUploaderInputReturn => {
+  const { fileLimitMaxSize, fileQueueLimit, isMultiple, onError } = props;
+
+  const [isDragging, setDragging] = useState(false);
+
+  const { fileQueue, addToQueue, clearQueue } = useFileUploaderContext();
+
+  const getUpdatedFileName = (name: string): string => `attachment__${name.replace(/\./g, '_')}`;
+
+  const checkAllowedFileSize = (file: File) => {
+    if (file.size > fileLimitMaxSize) {
+      throw new Error(DEFAULT_ERROR_MESSAGE_MAX_FILE_SIZE);
+    }
+  };
+
+  const checkFileDuplicity = (file: File) => {
+    if (isMultiple && fileQueue.has(getUpdatedFileName(file.name))) {
+      throw new Error(DEFAULT_ERROR_MESSAGE_QUEUE_DUPLICITY);
+    }
+  };
+
+  const checkQueueSize = () => {
+    if (fileQueue.size >= fileQueueLimit) {
+      throw new Error(DEFAULT_ERROR_MESSAGE_QUEUE_LIMIT);
+    }
+  };
+
+  const clearQueueHandler = () => {
+    if (!isMultiple) {
+      clearQueue();
+    }
+  };
+
+  const fileProcessHandler = (file: File) => {
+    try {
+      clearQueueHandler();
+      // TODO checkAllowedFormat(file);
+      checkFileDuplicity(file);
+      checkAllowedFileSize(file);
+      checkQueueSize();
+      addToQueue(getUpdatedFileName(file.name), file);
+    } catch (error) {
+      if (onError) {
+        onError(error);
+      } else {
+        console.warn(error);
+      }
+    }
+  };
+
+  const onDragOverHandler = (event: DragEvent<HTMLDivElement>) => {
```

Comment:
As I suggested earlier, this hook handles more things than only one. I strongly suggest splitting the dragging functionality into a more general `useDragAndDrop` hook and the file queue into a `useFileQueue` hook. This hook then can use both and the code would be more readable and maintainable. Also, I think that drag and drop hook will be useful somewhere else in the future.

Reply:
So I created a global hook `useDragAndDrop` and it is used here and can be used anywhere else at the same time